emperor/support_files/js/sceneplotview3d.js

```diff
@@ -219,6 +214,28 @@ define([
 
   /**
    *
+   * Reloads the plottable objects for all the decompostion views.
+   *
+   */
+  ScenePlotView3D.prototype.refresh = function() {
+
+    // Note that the internal logic of the THREE.Scene object prevents the
+    // objects from being re-added so we can simply iterate over all the
+    // decomposition views.
+
+    // Add all the meshes to the scene, iterate through all keys in
+    // decomposition view dictionary
+    for (var decViewName in this.decViews) {
+      for (var j = 0; j < this.decViews[decViewName].markers.length; j++) {
+        this.scene.add(this.decViews[decViewName].markers[j]);
+      }
+    }
+
+    this.needsUpdate = true;
```

Comment:
Good point, I've renamed the method to something clearer.  Note that we use the `needsUpdate` property to know when we have to render a new frame. Hence in this case when we add the objects to the scene, we notify the renderer that a new frame should be drawn on screen.